### PR TITLE
PR for issue #57

### LINKED ISSUE #57: Fixing player api endpoints
Post player info and avatar need to be fixed

### PR DESCRIPTION
**Pull request by GowthertG**
This pull request wil resolve #57

### CODE REVIEW

File: services/player/api/models.py [hunk 1]
```diff
@@ -6,9 +6,10 @@ class Player(AbstractUser):
     email = models.EmailField(max_length=30, blank=False, null=False)
     first_name = models.CharField(max_length=20, blank=False, null=False)
     last_name = models.CharField(max_length=20, blank=False, null=False)
-    tournament_name = models.CharField(max_length=20, blank=False, null=True)
+    tournament_name = models.CharField(max_length=20, blank=True, null=True)
```
Comment (zwzone) [inline services/player/api/models.py:9]: blank parameter must be False

File: services/player/api/views.py [hunk 2]
```diff
@@ -48,53 +52,33 @@ def get(self, request):
...
+                username = player['username']
+                first_name = player['first_name']
+                last_name = player['last_name']
+                avatar = player['avatar']
+            player, created = Player.objects.get_or_create(email=email, username=username, first_name=first_name, last_name=last_name, avatar=avatar)
+            if created:
                 return Response({
-                    "status": 400,
-                    "message": "Last name not provided",
+                    "status": 200,
```
Comment (zwzone) [inline services/player/api/views.py:73]: This should be 201 status code

File: services/player/api/views.py [hunk 3]
```diff
@@ -48,53 +52,33 @@ def get(self, request):
...
+                    "message": "User created successfully",
+                    "id": player.id
                 })
-            if not tournament_name:
+            else:
                 return Response({
                     "status": 400,
```
Comment (zwzone) [inline services/player/api/views.py:79]: This should be 200 status code

File: services/player/api/views.py [hunk 4]
```diff
@@ -288,6 +272,48 @@ def get(self, request):
                 "status": 500,
                 "message": "what" + str(e),
             })
```
Comment (zwzone) [inline services/player/api/views.py:274]: Missing new line here

**Review by reogine — CHANGES_REQUESTED** [review]
read zwzone comments

**Review by zwzone — CHANGES_REQUESTED** [review]
You need to change some lines in your code.

File: services/player/api/serializers.py [hunk 5]
```diff
@@ -13,15 +13,23 @@ class Meta:
...
         fields = ('username',)
 
+
 class FirstNameSerializer(serializers.ModelSerializer):
     class Meta:
         model = Player
         fields = ('first_name',)
-        
+
+
```
Comment (reogine) [inline services/player/api/serializers.py:22]: unnecessary new line

File: services/player/api/serializers.py [hunk 6]
```diff
@@ -13,15 +13,23 @@ class Meta:
...
 class LastNameSerializer(serializers.ModelSerializer):
     class Meta:
         model = Player
         fields = ('last_name',)
+
```
Comment (reogine) [inline services/player/api/serializers.py:27]: unnecessary new line

**Review by reogine — APPROVED** [review]
Some unnecessary new lines in the code.

File: services/player/api/models.py [hunk 7]
```diff
@@ -6,9 +6,9 @@ class Player(AbstractUser):
     email = models.EmailField(max_length=30, blank=False, null=False)
     first_name = models.CharField(max_length=20, blank=False, null=False)
     last_name = models.CharField(max_length=20, blank=False, null=False)
-    tournament_name = models.CharField(max_length=20, blank=False, null=True)
     username = models.CharField(max_length=20, blank=False, null=False, unique=True)
     avatar = models.URLField(blank=True, null=True)
+    id = models.AutoField(primary_key=True)
```
Comment (zwzone) [inline services/player/api/models.py:11]: I think this variable is already in the AbstractUser class

File: services/player/api/serializers.py (hunk 5 above)
Comment (zwzone) [reply services/player/api/serializers.py:22]: **Two** new lines between classes, and **One** new line between functions.

Comment (zwzone) [reply services/player/api/serializers.py:22]: Which means Two new lines is correct

File: services/player/api/serializers.py (hunk 6 above)
Comment (zwzone) [reply services/player/api/serializers.py:27]: The same thing here, two new lines are correct.

File: services/player/api/views.py (hunk 4 above)
Comment (zwzone) [reply services/player/api/views.py:272]: Missing new line here again

**Review by zwzone — APPROVED** [review]

**Pull request merged**